video.js

```diff
@@ -0,0 +1,90 @@
+import { onBindFirstLoad } from '../utils/jQueryBindFirst';
+import jQueryAjaxStop from '../utils/jQueryAjaxStop';
+import findElements from '../utils/findElements';
+import debounce from '../utils/debounce';
+
+const videoratio = 0.5625;
+
+export function setBackgroundImage(element, imageUrl) {
+  // Don't simply set "background:url(...)..." because this prop would override
+  // custom styling such as "background-size: cover".
+  element.setAttribute('style', `background-image:url(${imageUrl});background-color:#000;background-position:center center;background-repeat:no-repeat;`);
+}
+
+const debouncedResize = debounce(() => {
+  findElements('.container-lazyload').forEach((domContainerItem) => {
+    /* change ratio */
+    const embedelement = domContainerItem;
+    let newvideoratio = 0.5625;
```

Comment:
Instead of having `const videoratio = 0.5625;` AND `let newvideoratio = 0.5625;`, just have one of those variables. `videoratio` isn't even used anymore, so you should remove it and only use `newvideoratio` (and rename it to `videoratio`).

However, what you really want is not a `let` variable but a function like the following because you won't have to override variables and it'll help you to make the code more readable and reusable:

```
function determineVideoRatio(element) {
    const parent = element && element.parentNode && element.parentNode.parentNode;
    const hasAspectRatioClass = parent && parent.classList.contains('wp-has-aspect-ratio');

    if (hasAspectRatioClass) {
      const classes = String(parent.classList);
      const ratioclass = classes.substring(
        classes.lastIndexOf('wp-embed-aspect-'),
        classes.lastIndexOf(' '),
      );

      const ratioraw = ratioclass.replace('wp-embed-aspect-', '');
      const splitratio = ratioraw.split('-');

      return Number(splitratio[1]) / Number(splitratio[0]);
    }

    return 0.5625; // <-- default
}

...

const videoRatio = determineVideoRatio(embedelement);

...

const height = Math.round(width * videoRatio);
```

Bonus: Now that we have a function, it'll be easier to write tests for this. I'll write those tests once your changes are polished.

Reply:
Thanks for the feedback. I'll apply sugested changes as soon as possible. :-)